js/sparkswap.js

```diff
@@ -0,0 +1,544 @@
+            orders.push ({
+                'id': rawOrders[i].block_order_id,
+                'datetime': millisecondDatetime,
+                'timestamp': millisecondTimestamp,
+                'status': status,
+                'symbol': rawOrders[i].market,
+                'type': type,
+                'side': side,
+                'price': (this.safeFloat (rawOrders[i], 'amount') * this.safeFloat (rawOrders[i], 'limit_price')).toFixed (16),
+                'amount': this.safeFloat (rawOrders[i], 'amount'),
+                'info': rawOrders[i],
+            });
+        }
+        return orders;
+    }
+
+    async fetchTicker (symbol, params = {}) {
+        await this.loadMarkets ();
+        // The call to v1/market_stats will return the correct information that CCXT will
+        // expect, however we need to modify the payload to convert nanosecond timestamps
+        // to miliseconds
+        const res = await this.privateGetV1MarketStats ({ 'market': this.marketId (symbol) });
+        const millisecondTimestamp = this.nanoToMillisecondTimestamp (res.timestamp);
+        const millisecondDatetime = this.nanoToMillisecondDatetime (res.datetime);
+        return {
+            'symbol': res.symbol,
+            'timestamp': millisecondTimestamp,
+            'datetime': millisecondDatetime,
+            'high': res.high,
+            'low': res.low,
+            'bid': res.bid,
+            'bidVolume': res.bidVolume,
+            'ask': res.ask,
+            'askVolume': res.askVolume,
+            'vwap': res.vwap,
+            'baseVolume': res.baseVolume,
+            'counterVolume': res.counterVolume,
+        };
+    }
+
+    async fetchTrades (symbol, since = undefined, limit = undefined, params = {}) {
+        const request = {
+            'market': symbol,
+        };
+        if (since) {
+            request['since'] = since;
+        }
+        if (limit) {
+            request['limit'] = limit;
+        }
+        let response = await this.privateGetV1Trades (request);
+        let trades = response['trades'];
+        let formattedTrades = [];
+        for (let i = 0; i < trades.length; i++) {
+            formattedTrades.push ({
+                'id': trades[i].id,
+                'timestamp': this.nanoToMillisecondTimestamp (trades[i].timestamp),
+                'datetime': this.nanoToMillisecondDatetime (trades[i].datetime),
+                'order': trades[i].order,
+                'symbol': trades[i].market,
+                'type': trades[i].type,
+                'side': trades[i].side,
+                'price': trades[i].price,
+                'amount': trades[i].amount,
+                'info': trades[i],
+            });
+        }
+        return formattedTrades;
+    }
+
+    async withdraw (code, amount, address) {
+        const response = await this.privatePostV1WalletWithdraw ({
+            'symbol': code.toString (),
+            'amount': amount.toString (),
```

Comment:
We don't use `.toString` when dealing with amounts as it may lead to scientific notation in other languages. Please use this.decimalToPrecision instead.